tests/python/contrib/test_ethosu/test_codegen.py

```diff
@@ -1167,5 +1167,24 @@ def leaky_relu_func(x):
     _compare_tvm_with_tflite(leaky_relu_func, [ifm_shape], accel_type)
 
 
+@pytest.mark.parametrize("accel_type", ACCEL_TYPES)
+@pytest.mark.parametrize("units", [32, 64])
+@pytest.mark.parametrize("use_bias", [True, False])
+@pytest.mark.parametrize("activation_function", ["RELU", "NONE"])
+def test_tflite_fully_connected(
+    accel_type,
+    units,
+    use_bias,
+    activation_function,
+):
+    @tf.function
+    def fully_connected():
+        return tf.keras.layers.Dense(
```

Comment:
I'm not too familiar with the Keras API, but I'm not sure this will work. One thing we could do instead is use `tf.matmul` which gets legalized in the front-end to `qnn.dense` under the conditions we will use it for. e.g. something like this would be a starting point:
```
@tf.function
    def dense_layer(x):
        w = tf.constant(
            np.random.uniform(size=[units, units]),
            dtype=tf.float32,
        )
        return tf.matmul(x, w)

    _compare_tvm_with_tflite(dense_layer, [(1, units)], accel_type)
```

Happy to keep the Keras implementation if we get it working though, just wanted to offer an alternative :)

Reply:
this is a very welcome change, I'l try and make this work!